### ISSUE TITLE
Outgoing webhook not impersonating user

### ISSUE THREAD

**Issue by tensor5g** [RocketChat/Rocket.Chat#24442]
### Description:

With 4.4.0, responses to outgoing webhooks were broken entirely. Now with 4.4.1, it seems that they process the response properly, but the user the response is sent by is rocket.chat even with impersonation enabled

### Steps to reproduce:

1. Create simple outgoing webhook with impersonation enabled and user set to root
2. trigger outgoing webhook
3. observe that the response is sent by rocket.chat

### Expected behavior:

the response should be sent by the user who triggered the outgoing webhook

### Actual behavior:

rocket.cat posts the response

### Server Setup Information:

- Version of Rocket.Chat Server: 4.4.1
- Operating System: Docker
- Deployment Method: Docker
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: 
- MongoDB Version: 4.4

### Client Setup Information

- Desktop App or Browser Version: Latest Firefox/Edge tested
- Operating System: Windows 10

### Relevant logs:

I did not see any errors in server logs or browser console logs

**Comment by tensor5g** [comment]
I referred to the rocket.cat as rocket.chat a couple times by mistake. The user the response is being sent by is definitely rocket.cat

**Comment by sampaiodiego** [comment]
is this the behavior on 4.3? if so, can you please send the payload response it used to have this behavior? thanks

**Comment by tensor5g** [comment]
No, this was not the behavior prior to 4.4.0. A simple response payload such as {"text": "test"} will not be impersonated when before it would.

**Comment by tensor5g** [comment]
Issue is still present in 4.4.2. Another thing to mention is that no matter what user is selected for "post as", it will always display as rocket.cat unless impersonation is disabled, then it will post as the "post as" user.

**Comment by drdelaney** [comment]
I can confirm that this issue stopped working in 4.4.0, and was STILL broken in 5.3.6.
After spending multiple hours upgrading our old system from v2 to v5, I had to downgrade all the way back to 4.3.3 to test this.

**Comment by tensor5g** [comment]
Basically, instead of responding to the webhook with the message, you use whatever it is that received the webhook to interact directly with rocket.chat's API and post a message impersonating the user. 

use https://developer.rocket.chat/reference/api/rest-api/endpoints/core-endpoints/users-endpoints/create-users-token to get that user's context and then
https://developer.rocket.chat/reference/api/rest-api/endpoints/core-endpoints/chat-endpoints/send-message
to send the message using X-Auth-Token and X-User-Id headers

**Comment by drdelaney** [comment]
Will look into this.  thanks for the reply!
hopefully they can fix this upstream someday.
